src/Symfony/Bridge/PhpUnit/Legacy/SymfonyTestsListenerTrait.php

```diff
@@ -47,6 +47,8 @@ public function __construct(array $mockedNamespaces = array())
     {
         if (class_exists('PHPUnit_Util_Blacklist')) {
             \PHPUnit_Util_Blacklist::$blacklistedClassNames['\Symfony\Bridge\PhpUnit\Legacy\SymfonyTestsListenerTrait'] = 2;
+        } elseif (method_exists(Blacklist::class, 'addDirectory')) {
+            Blacklist::addDirectory('\Symfony\Bridge\PhpUnit\Legacy\SymfonyTestsListenerTrait');
```

Comment:
oh, wait, this needs a directory, not a class name!
also, we need to be sure that `Blacklist::initialize()` ran before (eg by calling it)
same in the bootstrap file

Reply:
if phpunit 9.1.3 is not yet released, maybe changing addDirectory to addClass($class, $level) is an idea? (and ensure it does the initialization also?)

Reply:
it is, see https://packagist.org/packages/phpunit/phpunit#9.1.3

Reply:
oops - could be worth adding still for 9.1.4 if Sebatian is OK

Reply:
and, anyway, `initialize` method is private :disappointed:

Reply:
`getBlacklistedDirectories()` is public and calls it without any other side effect

Reply:
Good. So, the only question left is how to add directory instead of class name. Should we use the same strategy used in `initialize` method (using `ReflectionClass`)?

Reply:
yep, inline I suppose, e.g.
`dirname(dirname((new \ReflectionClass(...))->getFileName())`

Reply:
Still, it's not clear to me how we can call non-static method `getBlacklistedDirectories` before calling static method `addDirectory`

Reply:
`(new BlackList())->getBlacklistedDirectories()` would do it, this is static state in the end

Reply:
Thank you Nicolas. Let me know if it's good for you now.